lib/vdc/DataMgr.cpp

```diff
@@ -547,11 +547,18 @@ bool is_int(std::string str) {
 
 template <class T>
 void _sanitizeFloats(T *buffer, size_t n) {
-	for (size_t i=0; i<n; i++) {
+	for (size_t i = 0; i < n; i++) {
 		if (isnan(buffer[i])) buffer[i] = std::numeric_limits<T>::infinity();
 	}
 }
 
+// MSVC has a bug where isnan() is not overloaded for integral types.
+// This function specializes the template to bypass this bug.
+template <>
+void _sanitizeFloats(int *buffer, size_t n) {
+	return;
+}
+
```

Comment:
I don't quite understand --- this function is called `_sanitizeFloats `, why does it have to handle integral types?
What's exactly the error Visual Studio throws if there's no this addition?

Reply:
It's a templated function, so it configures its specializations at compile time.

Since it can technically be passed an integral type, this specialization gets expanded and fails to compile.

MSVC not-so-helpfully reports the following error:
C2668 'fpclassify': ambiguous call to overloaded function

Strangely, this is at line 414 of math.h.  I'm not sure where it's being included.  The only files in our code base that include math.h are LegacyVectorMath.h and kdtree.c.

Reply:
Let's look at together on a windows machine. I'm not convinced that you create an empty function is a good fix to a potential programming error.